### PR TITLE
Add IronBank mint, redeem, borrow, repayBorrow and liquidation on Optimism

### PR DESCRIPTION
**Pull request by mmibmm**
Create tables by events on Optimism.

cc @jeff-dude 

Brief comments on the purpose of your changes:

*For Dune Engine V2*
I've checked that:
General checks:
* [X] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [X] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [X] if adding a new model, I added a test
* [X] the filename is unique and ends with .sql
* [X] each sql file is a select statement and has only one view, table or function defined  
* [X] column names are `lowercase_snake_cased`
* [X] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [X] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributors)
* [X] if adding a new materialized table, I edited the optimize table macro

Join logic:
* [X] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

Incremental logic:
* [X] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [X] where block_time >= date_trunc("day", now() - interval '1 week')
* [X] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [X] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: models/ironbank/optimism/ironbank_optimism_borrow.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+evt_block_time AS block_time,
+evt_tx_hash AS tx_hash,
+evt_index,
+borrower,
+i.underlying_token_address AS asset_address,
+borrowAmount AS borrow_amount
+FROM (
+    SELECT * FROM {{ source('ironbank_optimism', 'CErc20Delegator_evt_Borrow') }}
+) events
+LEFT JOIN ironbank_optimism.itokens i ON events.contract_address = i.contract_address
```
Comment (jeff-dude) [inline models/ironbank/optimism/ironbank_optimism_borrow.sql:22]: please replace all instances of ```ironbank_optimism.itokens``` with the jinja syntax for model reference (similar to sources):
{{ ref( <model_name> ) }}

File: models/ironbank/optimism/ironbank_optimism_sources.yml [hunk 2]
```diff
@@ -7,31 +7,108 @@ sources:
       warn_after: { count: 12, period: hour }
```
Comment (jeff-dude) [inline models/ironbank/optimism/ironbank_optimism_sources.yml:7]: can you do me a favor and run the dbt source freshness command on all sources here? something like this (or close) should work:
```dbt source freshness --select +models/ironbank/optimism/```

if any return warnings or errors, they may need to either expand their timeframe or be removed all together

i will take note to try and include this in the gh action, so it's included as an automation test

**Review by jeff-dude — CHANGES_REQUESTED** [review]
thanks @mmibmm for the follow-up spell. two minor pieces of feedback

File: models/ironbank/optimism/ironbank_optimism_borrow.sql (hunk 1 above)
Comment (mmibmm) [reply models/ironbank/optimism/ironbank_optimism_borrow.sql]: Sure~ fixed.

File: models/ironbank/optimism/ironbank_optimism_sources.yml (hunk 2 above)
Comment (mmibmm) [reply models/ironbank/optimism/ironbank_optimism_sources.yml]: I got some error and I chose to remove them all.

**Review by jeff-dude — APPROVED** [review]
thanks again @mmibmm -- note that views will be built once per day, you should see no later than tomorrow

**Pull request merged**